### PR TITLE
[WIP][mysql] Add ExternalSystemListener to notify user about the status of connector.

### PR DESCRIPTION
**Pull request by lvyanquan**
this closes https://github.com/ververica/flink-cdc-connectors/issues/2520

Try to support notification about the status of connector, since it may be a common demand from community.
However, I met some problem when add a preliminary implementation, want to get some advice from yours.
1. When I try to define an enum of ExternalSystemEvent, I found that it's similar to [AssignerStatus](https://github.com/ververica/flink-cdc-connectors/blob/6ca87c4d57a790c0e51247b275f82c03d8a8f9f8/flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/AssignerStatus.java#L117), can we reuse it?
2. Should we send information about finished splits to listener? then it would be very similar to checkpoint.
3. Should we add a default implement like KafkaListener?

### CODE REVIEW

**Comment by lvyanquan** [comment]
Hi, @ruanhang1993 
Not sure if my design and confusion are reasonable, Looking forward to receiving your opinion.

**Comment by SML0127** [comment]
@lvyanquan
Wow..
First of all, Thank you for implementing this function I was hoping for.

I've been so busy at work and was just about to implement it, but you've already implemented it! 🤣🤣🤣

Can I run this pr with our company in-house notification tools and give you some feedback??🙇🏻‍♂️🙇🏻‍♂️

**Comment by SML0127** [comment]
> @lvyanquan Wow.. First of all, Thank you for implementing this function I was hoping for.
> 
> I've been so busy at work and was just about to implement it, but you've already implemented it! 🤣🤣🤣
> 
> Can I run this pr with our company in-house notification tools and give you some comment??🙇🏻‍♂️🙇🏻‍♂️

I added some information (e.g. `startupMode`, `gtids`) for notification message and created @[PR](https://github.com/lvyanquan/flink-cdc-connectors/pull/2).

As our team implemented and used this notification feature, we realized that the `startupMode` and `gtids` information was really useful. So I added it.

I hope you give it a try, and thanks for any comments!! 😄🙇🏻‍♂️

**Comment by lvyanquan** [comment]
It makes sense to me, and I create an abstract class [AbstractListenerMessage](https://github.com/ververica/flink-cdc-connectors/pull/2523/files#diff-698c27eee7807422a2eee230a778860d120669f04191d4dc28432cf0614190f4) since there are other datasources like oracle、postgreSQL.

**Comment by SML0127** [comment]
> It makes sense to me, and I create an abstract class [AbstractListenerMessage](https://github.com/ververica/flink-cdc-connectors/pull/2523/files#diff-698c27eee7807422a2eee230a778860d120669f04191d4dc28432cf0614190f4) since there are other datasources like oracle、postgreSQL.

LGTM!
I learn a lot from your implementation😃

File: flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/listener/AbstractListenerMessage.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.Properties;
+
+/** The information that will be sent with the notification. */
+public abstract class AbstractListenerMessage {
+
+    public Properties getProperties() {
+        return properties;
+    }
+
+    Properties properties = new Properties();
```
Comment (whhe) [inline flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/listener/AbstractListenerMessage.java:28]: This field 'properties' is for?

**Comment by yuxiqian** [comment]
Is there any progress on this PR? cc @lvyanquan

**Pull request closed without merging**